### ISSUE TITLE
[Feature Request] contact sharing (share vCard contacts VCF)

### ISSUE THREAD

**Issue by mdan312** [telegramdesktop/tdesktop#4696]
Here are some feature suggestions for Contact sharing. (Some suggestions are not relevant for the "desktop" app, please let me know where I can report an issue for Android / iOS)

1. When asking to share a contact from pressing the paperclip icon: should read the entire VCARD and ask what we want to share (cf. WhatsApp)
2. The user_id should be added when the shared contact uses Telegram (it is not done yet!!!) => impossible to interact from Telegram!! 
3. A button "Add contact to Address book" should be there.
4. share a vCard from AdressBook (or from a file) with telegram : should be possible (include email, ...) and there should be only one contact, not many. (try do to it: it is buggy!!)

Telegram has a very poor implementation of that feature compared to WhatsApp. Should be a priority as it restrain people to interact through Telegram.

**Comment by mdan312** [comment]
share a vCard with telegram  does not work. Sends many times the same contact: one for each phone number  instead of one vcard. Does not include the address nor any other info

**Comment by stek29** [comment]
Because Telegram API only has 3 fields for sent contacts: phone number, first name and last name: https://stek29.rocks/tl-schema/latest/constructor/inputMediaContact

Speaking of "no link with contacts": contact -> user mapping is done on server side: Client sends phone number and name, [server adds user_id if it exists](https://stek29.rocks/tl-schema/latest/constructor/messageMediaContact) and also sends full user info whenever that message is requested in users field of messages.messages.

So, Android app has the best behavior possible in such situation. 

For TDesktop -- I don't think it has any vcard parsing at all -- so vcf file would be sent as is -- as file and not as contact.

Also, most of issues you described reference address book, so I assume they're mostly for Android app, and this is not a proper place for them.

So, to recap, only *valid* (TDesktop) issue I see is lack of vcf file handling in TDesktop (could be done as images -- send as file or send as ~~photo~~ contact)

**Comment by mdan312** [comment]
> A valid (TDesktop) issue I see is lack of vcf file handling in TDesktop (could be done as images -- send as file or send as contact)

Yes it would be good.

**Comment by MasterGroosha** [comment]
Is it still valid? Seems that Android/iOS versions already support vCard.

**Comment by Aokromes** [comment]
near 6 months without answer, if this request is still valid plz ask to reopen.
